src/qofonosimwatcher.cpp

```diff
@@ -150,7 +161,8 @@ QOfonoSimWatcher::QOfonoSimWatcher(QObject *parent) :
 
 QOfonoSimWatcher::~QOfonoSimWatcher()
 {
-    // d_ptr is a child object, it gets deleted automatically
```

Comment:
You could possibly fix the problem by simply explicitly deleting the private object here; normally it would get deleted in the QObject dtor, but perhaps there is a window where event processing is occurring between the start of ~QOfonoSimWatcher() and the end of ~QObject()?